### PR TITLE
A minor change for transforms.RandomCrop

### PR DESCRIPTION
**Pull request by anguelos**
I just added an optional parameter that pads an image if needed so that small errors don't raise exceptions.
The option is by default turned off so that API behavior is exactly the same.
As far as I saw the test cases don't test for Exceptions so i did not add a test for the feature being off but I tested that locally.

### CODE REVIEW

File: test/test_transforms.py [hunk 1]
```diff
@@ -1164,4 +1174,4 @@ def test_random_grayscale(self):
 
 
 if __name__ == '__main__':
-    unittest.main()
+    unittest.main()
```
Comment (fmassa) [inline test/test_transforms.py]: lint is complaining because of no newline in the end of the file

File: torchvision/transforms/transforms.py [hunk 2]
```diff
@@ -408,6 +411,13 @@ def __call__(self, img):
         if self.padding > 0:
             img = F.pad(img, self.padding)
 
+        # pad the width if needed
+        if self.pad_if_needed and img.size[0] < self.size[1]:
+            img = F.pad(img, (int((1 + self.size[1] - img.size[0]) / 2), 0))
+        # pad the height if needed
+        if self.pad_if_needed and img.size[1] < self.size[0]:
+            img = F.pad(img,(0, int((1 + self.size[0] - img.size[1]) / 2)))
```
Comment (fmassa) [inline torchvision/transforms/transforms.py]: lint is complaining about space missing after the first `,`

**Comment by anguelos** [comment]
The context is texture analysis.
Scaling samples with resize would distort texture properties.
I use a dataset were I want during inference to have variable sized images.
But train on randomly cropped fixed size images so that they are easy to collate. 
As the situation was I had to set the cropping size to be smaller than the smallest sample in the dataset.

**Comment by anguelos** [comment]
I have updated the two linting errors. The Travis failure appears to not be related with my pull request.

**Pull request merged**